packages/pangea-node-sdk/src/request.ts

```diff
@@ -11,27 +11,41 @@ const delay = async (ms: number) =>
     setTimeout(resolve, ms);
   });
 
+interface request extends Object {
+  config_id?: string;
+}
+
 class PangeaRequest {
   private serviceName: string;
   private token: string;
   private config: PangeaConfig;
   private extraHeaders: Object;
-  private userAgent: string;
-
-  constructor(serviceName: string, token: string, config: PangeaConfig) {
+  private checkConfigID: boolean;
+  private userAgent: string = "";
+
+  constructor(
+    serviceName: string,
+    token: string,
+    config: PangeaConfig,
+    checkConfigID: boolean = false
+  ) {
     if (!serviceName) throw new Error("A serviceName is required");
     if (!token) throw new Error("A token is required");
 
     this.serviceName = serviceName;
     this.token = token;
     this.config = new PangeaConfig({ ...config });
-    this.userAgent = "";
     this.setCustomUserAgent(config.customUserAgent);
     this.extraHeaders = {};
+    this.checkConfigID = checkConfigID;
   }
 
-  async post(endpoint: string, data: object): Promise<PangeaResponse<any>> {
+  async post(endpoint: string, data: request): Promise<PangeaResponse<any>> {
     const url = this.getUrl(endpoint);
+    if (this.checkConfigID && this.config.configID && data.config_id === undefined) {
```

Comment:
I'm not sure I see why checkConfigId is needed.

I feel like (this.config.configID && data.config_id === undefined) should be enough. I guess you are using it as a way for every other service to not use config_id?

Is PangeaConfig a class or just an interface? I wonder if maybe you could just include isMultiConfig as a control param on the PangeaConfig object, and then only Audit is setting it true, that way every other service doesn't have to worry about this param

I bring this up, only because often positional arguments in javascript can get annoying, so I think if "checkConfigId" can b just apart of the PangeaConfig, that could be nicer. But this comment is just kind of a brain dump, so take it or leave it

Reply:
I use this flag to setup config id use per service. So if service should not support config ID I don't check it even if user set up it on PangeaConfig.
PangeaConfig is a class and it's initialized with an interface.
I don't like the idea of isMultiConfig flag inside PangeaConfig due to this class is left to user settings and user should not choose if service is or is not multiConfig. 
checkConfigId and supportMultiConfig are flags that we set to true internally only in audit service right now, user should no change them.

Reply:
I understand not wanting to extend PangeaConfig, if the idea is it should be full user controlled.

I think if this is internal only, then it shouldn't be apart of the constructor then for BaseService because it is not dyanmic

```
current:

class BaseService {
  private supportMultiConfig: boolean;

  constructor(...args, supportMultiConfig: boolean) {

  }
}
```

Maybe
```
class BaseService {
  private isMultiConfigSupported: boolean = false;

  constructor(...args) {

  }
}

class AuditService extends BaseService {
  // Should just be able to change the private field here, I believe?
  private isMultiConfigSupported: boolean = true;
}

// Then every other service doesn't need to change anything
```

I see you are calling the constructor for PangeaRequest though, so the only thing I would say is could we use the same variable name, isMultiConfigSupported in both cases, and instead of a positional options as the last positional argument, maybe just have a `options: { isMultiConfigSupported: boolean }`, just to make it a bit simpler to add more optional params letter

Reply:
Sure. I like what you propose, I'll implement that. Thanks for feedback!

Reply:
I remember... the problem doing that way, I mean, having isMultiConfigSupported flag outside constructor is that I'll set it to true after BaseService constructor execution, and PangeaRequest constructor is executed inside BaseService constructor, so PangeaRequest constructor would execute with that flag always with false value. Any other idea? 
Anyway I can pass isMultiConfigSupported as options to BaseConstructor too, if you agree with that

Reply:
PangeaRequest constructor is executed inside BaseService constructor, so PangeaRequest constructor would execute with that flag always with false value. Any other idea? -> This isn't true I don't believe

Reply:
If you override the private variable default value in the child class (AuditService) the BaseService constructor will still reference what the private variable is set to

Reply:
Maybe my C++ background is playing a trick on me or maybe I'm just wrong. I'll give it a try and let you know!